hooks.lua

```diff
@@ -9,14 +9,47 @@ local b64e = ie.require("mime").b64
 irc.hooks = {}
 irc.registered_hooks = {}
 
+local colour_codes = {
+	[0] = "white",
+	[1] = "black",
+	[2] = "blue",
+	[3] = "green",
+	[4] = "red",
+	[5] = "darkred",
+	[6] = "purple",
+	[7] = "orange",
+	[8] = "yellow",
+	[9] = "limegreen",
+	[10] = "teal",
+	[11] = "cyan",
+	[12] = "deepskyblue",
+	[13] = "pink",
+	[14] = "grey",
+	[15] = "lightgrey",
+	[99] = "#ffffff"
+}
+
+local function convert_colour(text)
+	local lines = text:split("\3", true)
+	for index, line in pairs(lines) do
+		local irc_code = line:match("^(%d+)")
```

Comment:
This block would take e.g. "01foo" at the start of the string as an escape sequence.